python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = False
+
+
+class CrosstalkCoeffMock(IsrMock):
+    r"""Simulated crosstalk coefficient matrix.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = True
+        self.config.doTransmissionCurve = False
+
+
+class TransmissionMock(IsrMock):
+    r"""Simulated transmission curve.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = True
+
+
+class DataRefMock(object):
+    r"""Simulated gen2 butler data ref.
+
+    Currently only supports get and put operations, which are most
+    likely to be called for data in ISR processing.
+
+    """
+    dataId = "isrMock Fake Data"
+    darkval = 2.  # e-/sec
+    oscan = 250.  # DN
+    gradient = .10
+    exptime = 15  # seconds
+    darkexptime = 40.  # seconds
+
+    def __init__(self, **kwargs):
+        if 'config' in kwargs.keys():
+            self.config = kwargs['config']
+        else:
+            self.config = None
+
+    def get(self, dataType, **kwargs):
+        r"""Return an appropriate data product.
+
+        Parameters
+        ----------
+        dataType : `str`
+            Type of data product to return.
+
+        Returns
+        -------
+        mock : IsrMock.run() result
+            The output product.
+        """
+        if "_filename" in dataType:
+            return tempfile.mktemp(), "mock"
+        elif 'transmission_' in dataType:
+            return TransmissionMock(config=self.config).run()
+        elif dataType == 'ccdExposureId':
+            return 20090913
+        elif dataType == 'camera':
+            return IsrMock(config=self.config).getCamera()
+        elif dataType == 'raw':
+            return RawMock(config=self.config).run()
+        elif dataType == 'bias':
+            return BiasMock(config=self.config).run()
+        elif dataType == 'dark':
+            return DarkMock(config=self.config).run()
+        elif dataType == 'flat':
+            return FlatMock(config=self.config).run()
+        elif dataType == 'fringe':
+            return FringeMock(config=self.config).run()
+        elif dataType == 'defects':
+            return DefectMock(config=self.config).run()
+        elif dataType == 'bfKernel':
+            return BfKernelMock(config=self.config).run()
+        elif dataType == 'linearizer':
+            return None
+        elif dataType == 'crosstalkSources':
+            return None
+        else:
+            return None
```

Comment:
OK, in this situation I support the switch-case, this seems fine. However, could you return something worse on the fall-through case? Either raise to let someone know they asked for a thing which couldn't be `get()`'ed, or if not that for some reason, even just a string saying `"You asked for %s and I don't know how to give you one of those" % dataType`?

Reply:
OK, the latter suggestion is pretty dumb as it'll just end up raising one step down the line, but I do feel that falling through and silently giving `None` isn't great; it's already hard enough to debug things involving the butler.